test/software/amazon/ion/EquivsTestCase.java

```diff
@@ -187,12 +191,45 @@ protected void checkEquivalence(final IonValue left,
         }
     }
 
+    public static IonDatagram[] roundTripDatagram(IonDatagram input) throws IOException{
+        IonSystem system = IonSystemBuilder.standard().build();
+        IonLoader loader = system.getLoader();
+        ByteArrayOutputStream textOutputStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream binaryOutputStream = new ByteArrayOutputStream();
+        IonWriter textWriter = IonTextWriterBuilder.standard().build(textOutputStream);
+        IonWriter binaryWriter = IonBinaryWriterBuilder.standard().build(binaryOutputStream);
+        IonDatagram[] data = new IonDatagram[3];
+        try {
+            IonReader reader = system.newReader(input);
+            textWriter.writeValues(reader);
+            reader = system.newReader(input);
+            binaryWriter.writeValues(reader);
+
+        } finally {
+            textWriter.close();
+            binaryWriter.close();
+        }
+        data[0] = input;
+        data[1] = loader.load(new ByteArrayInputStream(textOutputStream.toByteArray()));
+        data[2] = loader.load(new ByteArrayInputStream(binaryOutputStream.toByteArray()));
+        return data;
+    }
+
+    public void roundTripEquivalence(IonDatagram input, boolean myExpectedEquality) throws IOException {
+        IonDatagram[] data = roundTripDatagram(input);
+        for(int i = 0; i < data.length - 1; i++){
+            runEquivalenceChecks(data[i], myExpectedEquality);
```

Comment:
The last element in the `data` array is never subjected to `runEquivalenceChecks`.